phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/ui/SearchBox.java

```diff
@@ -1,31 +1,34 @@
 package org.phoenicis.javafx.views.mainwindow.ui;
 
-import javafx.event.ActionEvent;
-import javafx.event.EventHandler;
-import org.phoenicis.javafx.views.common.ThemeManager;
-
 import javafx.beans.property.StringProperty;
-import javafx.scene.control.TextField;
 import javafx.scene.control.Button;
+import javafx.scene.control.TextField;
 import javafx.scene.layout.AnchorPane;
 
 import java.util.function.Consumer;
 
 public class SearchBox extends AnchorPane {
     private TextField searchField;
     private Button clearButton;
+    private Consumer<String> onSearch;
```

Comment:
No, I don't want to throw an error. I just want to log that something is not correct and I want this log in the constructor and not an exception every time the search is used. If we cannot implement a `onSearch` function, it doesn't make any sense to use the `SearchBox`. So I don't believe that this is a valid use case.